lib/response.js

```diff
@@ -56,8 +56,14 @@ var charsetRegExp = /;\s*charset\s*=/;
  * @public
  */
 
-res.status = function status(code) {
-  this.statusCode = code;
+res.status = function status(statusCode) {
+  // check that status code is valid
+  statusCode = parseInt(statusCode, 10);
```

Comment:
Also, I don't think we should parse at all. Throw if it is not a number primitive